src/shared/sam/debugger/goSamDebug.ts

```diff
@@ -89,8 +86,18 @@ export async function makeGoConfig(config: SamLaunchRequestArgs): Promise<GoDebu
         throw Error('missing launch.json, template.yaml, and failed to discover project root')
     }
 
-    let localRoot: string | undefined
-    let remoteRoot: string | undefined
+    // If the target is 'code' we need to adjust codeRoot to the location of the source
+    if (config.invokeTarget.target === 'code') {
```

Comment:
Can we instead require that the codeRoot/codeuri point to the subdirectory?

Reply:
Oh that's what it actually does, mis-wording there.

Reply:
I mean `projectRoot` in the launch config. IOW, can we avoid the bespoke handler format, by requiring the user to set `projectRoot` as we need it?

Reply:
The `projectRoot` points to the manifest's location for all (?) runtimes currently. SAM expects `CodeUri` in our fake template to point to the handler's entry point directory. Perhaps it would be better to change `projectRoot` when the launch config is created? But I'm not entirely sure what side-effects that would cause. I can try it though.

Reply:
> SAM expects `CodeUri` in our fake template to point to the handler's entry point directory. Perhaps it would be better to change `projectRoot` when the launch config is created?

That's what I'm thinking. Though it means projectRoot was a bad name (why didn't we choose "codeUri"...?). But it would be less cognitive overhead if we mapped to SAM expectations 1-to-1 where possible.

Reply:
I decided to make it so Go lambda handlers can be a path for launch configs. This should be much more intuitive for users then using some weird '.' or ':' scheme for resolving the source directory.